packages/webdriver/src/request/request.ts

```diff
@@ -0,0 +1,61 @@
+import { performance } from 'node:perf_hooks'
+import type { URL } from 'node:url'
+import dns from 'node:dns'
+
+import type { Options } from '@wdio/types'
+
+import WebDriverRequest, { RequestLibError } from './index.js'
+
+// As per this https://github.com/node-fetch/node-fetch/issues/1624#issuecomment-1407717012 we are setting ipv4first as default IP resolver.
+// This can be removed when we drop Node18 support.
+dns.setDefaultResultOrder('ipv4first')
+
+export default class FetchRequest extends WebDriverRequest {
+    constructor (method: string, endpoint: string, body?: Record<string, unknown>, isHubCommand: boolean = false) {
+        super(method, endpoint, body, isHubCommand)
+    }
+
+    protected async _libRequest (url: URL, opts: Options.RequestLibOptions) {
+        try {
+            let headers: any = { ...opts.headers }
+
+            if (opts.username && opts.password) {
+                const encodedAuth = Buffer.from(`${opts.username}:${opts.password}`, 'utf8').toString('base64')
+                headers = {
+                    ...headers,
+                    Authorization: `Basic ${encodedAuth}`,
+                }
+            }
+
+            const controller = new AbortController()
+            const timeoutId = setTimeout(() => controller.abort(), opts.timeout || 120000)
+
+            const response = await fetch(url, {
+                method: opts.method,
+                body: opts.body ?? JSON.stringify(opts.json),
+                headers,
+                signal: controller.signal,
+            })
+
+            // Cloning the response to prevent body unusable error
+            const resp = response.clone()
+
+            // Clear the timeout as the request completed successfully
+            clearTimeout(timeoutId)
+
+            return {
+                statusCode: resp.status,
+                body: await resp.json() ?? {},
+            } as Options.RequestLibResponse
+        } catch (err: any) {
```

Comment:
```suggestion
        } catch (err: unknown) {
```

Reply:
We use the `any` type here because we're checking if the error thrown is not an instance of the `Error`. If we switch to the `unknown` type, we'd need to redefine the typing to access the message property. Please suggest.

Reply:
Sounds good. We can disregard this.